x/nexus/types/types.go

```diff
@@ -188,3 +189,79 @@ func NewTransferEpoch(chain exported.ChainName, asset string, epoch uint64, dire
 		Direction: direction,
 	}
 }
+
+// CallContractProposalMinDeposits is a set of CallContractProposalMinDeposit, one per contract address
+type CallContractProposalMinDeposits []CallContractProposalMinDeposit
+
+type callContractProposalMinDepositsMap map[string]map[string]sdk.Coins
+
+// Get returns the minimum deposit for the given chain and contract address
+func (m callContractProposalMinDepositsMap) Get(chain exported.ChainName, contractAddress string) sdk.Coins {
+	c := strings.ToLower(chain.String())
+	address := strings.ToLower(contractAddress)
+
+	if _, ok := m[c]; !ok {
+		return sdk.Coins{}
+	}
+
+	return m[c][address]
+}
+
+// ValidateBasic returns an error if the type is invalid
+func (minDeposits CallContractProposalMinDeposits) ValidateBasic() error {
+	chainContractAddressPairs := make(map[string]bool)
```

Comment:
I'd use `struct{}` instead of bool because it's a zero cost abstraction, but that's a minor detail